ihe-interop/src/main/java/org/motechproject/ihe/interop/service/impl/IHETemplateServiceImpl.java

```diff
@@ -19,14 +21,41 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(IHETemplateServiceImpl.class);
 
+    private PostMethod post;
+
+    private HttpClient client;
```

Comment:
Maybe I'm missing something, but I see two methods here that immadiately call "setConnectionParameters", which recreates the PostMethod and HttpClient from scratch.